Use code snippets for code examples

nfnt opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://github.com/kudobuilder/kudo/blob/master/CONTRIBUTING.md
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://github.com/kudobuilder/kudo/blob/master/RELEASE.md
3. Ensure you have added or ran the appropriate tests for your PR
4. If the PR is unfinished, start it as a Draft PR: https://github.blog/2019-02-14-introducing-draft-pull-requests/
-->

**What this PR does / why we need it**:
To ensure that examples are in sync with KUDO, their code is taken directly from the KUDO repository. The KUDO repository is added as a submodule. The [markdown-it-vuepress-code-snippet-enhanced](https://github.com/fabbballin/markdown-it-vuepress-code-snippet-enhanced) plugin is used for this, because it supports viewing only a range of lines of a snippet.

**Which issue(s) this PR fixes**:
<!-- 
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes #31 

**Special notes for your reviewer**:

Commit:
Use code snippets for code examples

To ensure that examples are in sync with KUDO, their code is taken directly from the KUDO repository. The KUDO repository is added as a submodule. The [markdown-it-vuepress-code-snippet-enhanced](https://github.com/fabbballin/markdown-it-vuepress-code-snippet-enhanced) plugin is used for this, because it supports viewing only a range of lines of a snippet.

meichstedt commented on content/docs/developing-operators.md:
```diff
@@ -136,22 +76,11 @@ Plan foo
 Plans consists of one or more `phases`. `Phases` consists of one or more `steps`. `Steps` contain one or more `tasks` (those are defined in the section we talked about in the last paragraph). Both phases and also steps can be configured with an execution `strategy`, either `serial` or `parallel`.
 
-The sample has a `deploy` plan with a `deploy-phase` and a `deploy-step`. From the `deploy-step` the `deploy-task` is referenced. This task gets executed when an instance is created using the operator.
+The sample has a `deploy` plan with a `main` and a `everything`. From `everything` the `app` is referenced. This task gets executed when an instance is created using the operator.
```
```suggestion
The sample has a `deploy` plan with a phase `main` and a step `everything`. From `everything` the `app` is referenced. This task gets executed when an instance is created using the operator.
```

meichstedt replied:
One downside of including files is that someone might change IDs in the file which leads to weird documentation. I don't think we can prevent this though and the benefit of including them weighs heavier I think.

meichstedt commented on .gitmodules:
```diff
@@ -0,0 +1,3 @@
+[submodule "kudo"]
+	path = kudo
+	url = https://github.com/kudobuilder/kudo.git
```
Are we OK with master or would we want to pin a specific branch/tag?

nfnt replied:
It's pinned to `v0.6.0` which has a SHA of `c74d22743d6ec37e24e89eaeb64803597187fe08`.

meichstedt reviewed (approved):
Thanks @nfnt, that's pretty neat!

Do you know if the build will fail in case a file was moved in the kudo repo? I hope it does :)

kensipe committed:
Apply suggestions from code review

Co-Authored-By: Matthias Eichstedt <matthias.eichstedt@mesosphere.io>

Commit:
Merge branch 'master' into nfnt/import-code-snippets

Commit:
Only use snippets of complete files

This way we will get build failures in VuePress >=1.2.0 if snippet files aren't found.

Commit:
Use KUDO v0.7.5 for code snippets

Commit:
Update yarn.lock

Commit:
Remove unused plugin from configuration

meichstedt reviewed (approved):
`yarn docs:build` actually fails when sources aren't present:

```bash
error Code snippet path not found: /Users/matthias/Documents/projects/kudobuilder/www/kudo/config/samples/first-operator/operator.yaml
error Code snippet path not found: /Users/matthias/Documents/projects/kudobuilder/www/kudo/config/samples/first-operator/templates/deployment.yaml
error Code snippet path not found: /Users/matthias/Documents/projects/kudobuilder/www/kudo/config/samples/first-operator/params.yaml
Language does not exist text
wait Rendering static HTML...
wait vuepress-plugin-feed Adding pages/posts as feed items...
success vuepress-plugin-feed added 7 page(s) as feed item(s)
wait vuepress-plugin-feed Checking feeds that need to be generated...
success vuepress-plugin-feed rss2 feed file generated and saved to content/.vuepress/dist/rss.xml
success vuepress-plugin-feed atom1 feed file generated and saved to content/.vuepress/dist/feed.atom
success vuepress-plugin-feed json1 feed file generated and saved to content/.vuepress/dist/feed.json
success Generated static files in content/.vuepress/dist.

error Command failed with exit code 1.
```

Commit:
Merge branch 'master' into nfnt/import-code-snippets

Commit:
Update to KUDO v0.8.0

Commit:
Merge branch 'master' into nfnt/import-code-snippets

Commit:
Document KUDO submodule updates

meichstedt reviewed (approved):
Thanks for documenting the submodule approach @nfnt!
